app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutViewModel.java

```diff
@@ -22,22 +21,52 @@ public class JoinHangoutViewModel extends ViewModel {
 
     MutableLiveData<Errors> errors = new MutableLiveData();
     MutableLiveData<Hangout> toJoinHangout = new MutableLiveData<>();
+    MutableLiveData<Hangout> joinedHangout = new MutableLiveData<>();
 
-    public void joinHangout(Hangout hangout){
-        if(!userInHanogut(hangout)){
-
-        }else{
-            errors.postValue(Errors.USER_ALREADY_IN_HANGOUT);
+    public void joinHangout(Hangout hangout) {
+        if (!userInHanogut(hangout)) {
+            JSONArray updatedMembers = hangout.getMembers();
+            updatedMembers.put(ParseUser.getCurrentUser());
+            hangout.put(Hangout.KEY_MEMBERS, updatedMembers);
+            hangout.saveInBackground(new SaveCallback() {
+                @Override
+                public void done(ParseException e) {
+                    if (e != null) {
+                        errors.postValue(Errors.ERROR_UPDATING_HANGOUT);
+                    } else {
+                        addHangoutToUser(hangout);
+                    }
+                }
+            });
+        } else {
+            errors.postValue(Errors.ERROR_USER_ALREADY_IN_HANGOUT);
         }
     }
 
+    private void addHangoutToUser(Hangout hangout) {
```

Comment:
There might be room for improvement here. When the joining operation is made, the database needs two updates, one for the user object and one for the hangout object. But what would happen if for some reason one is successful and one fails. There would be inconsistencies in the database. I was thinking making a new table of a UserHangout relation so just one table needs updates when joining.